assets/js/blocks/featured-category/block.js

```diff
@@ -0,0 +1,397 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import {
+	AlignmentToolbar,
+	BlockControls,
+	InnerBlocks,
+	InspectorControls,
+	MediaUpload,
+	MediaUploadCheck,
+	PanelColorSettings,
+	withColors,
+} from '@wordpress/editor';
+import {
+	Button,
+	FocalPointPicker,
+	IconButton,
+	PanelBody,
+	Placeholder,
+	RangeControl,
+	ResizableBox,
+	Spinner,
+	ToggleControl,
+	Toolbar,
+	withSpokenMessages,
+} from '@wordpress/components';
+import classnames from 'classnames';
+import { Component, Fragment } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { debounce, isObject } from 'lodash';
+import PropTypes from 'prop-types';
+import { IconFolderStar } from '../../components/icons';
+
+/**
+ * Internal dependencies
+ */
+import ProductCategoryControl from '../../components/product-category-control';
+
+/**
+ * The min-height for the block content.
+ */
+const MIN_HEIGHT = wc_product_block_data.min_height;
+
+/**
+ * Get the src from a category object, unless null (no image).
+ *
+ * @param {object|null} category A product category object from the API.
+ * @return {string}
+ */
+function getCategoryImageSrc( category ) {
+	if ( isObject( category.image ) ) {
+		return category.image.src;
+	}
+	return '';
+}
+
+/**
+ * Get the attachment ID from a category object, unless null (no image).
+ *
+ * @param {object|null} category A product category object from the API.
+ * @return {int}
+ */
+function getCategoryImageID( category ) {
+	if ( isObject( category.image ) ) {
+		return category.image.id;
+	}
+	return 0;
+}
+
+/**
+ * Generate a style object given either a product category image from the API or URL to an image.
+ *
+ * @param {string} url An image URL.
+ * @return {object} A style object with a backgroundImage set (if a valid image is provided).
+ */
+function backgroundImageStyles( url ) {
+	if ( url ) {
+		return { backgroundImage: `url(${ url })` };
+	}
+	return {};
+}
+
+/**
+ * Convert the selected ratio to the correct background class.
+ *
+ * @param {number} ratio Selected opacity from 0 to 100.
+ * @return {string} The class name, if applicable (not used for ratio 0 or 50).
+ */
+function dimRatioToClass( ratio ) {
+	return ratio === 0 || ratio === 50 ?
+		null :
+		`has-background-dim-${ 10 * Math.round( ratio / 10 ) }`;
+}
+
+/**
+ * Component to handle edit mode of "Featured Product Category".
```

Comment:
In some places we are using the naming `Featured Category` and in some others `Featured Product Category`. I don't have any preference for one name or the other, but I think we should unify it to avoid confusions.